tests/components/eheimdigital/test_light.py

```diff
@@ -0,0 +1,198 @@
+"""Tests for the light module."""
+
+from unittest.mock import MagicMock, patch
+
+from eheimdigital.types import EheimDeviceType, LightMode
+import pytest
+from syrupy.assertion import SnapshotAssertion
+
+from homeassistant.components.eheimdigital.const import EFFECT_DAYCL_MODE
+from homeassistant.components.light import (
+    ATTR_BRIGHTNESS,
+    ATTR_EFFECT,
+    DOMAIN as LIGHT_DOMAIN,
+)
+from homeassistant.const import (
+    ATTR_ENTITY_ID,
+    SERVICE_TURN_OFF,
+    SERVICE_TURN_ON,
+    Platform,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import entity_registry as er
+
+from tests.common import MockConfigEntry, snapshot_platform
+
+
+@pytest.mark.parametrize(
+    "tankconfig",
+    [
+        [["CLASSIC_DAYLIGHT"], []],
+        [[], ["CLASSIC_DAYLIGHT"]],
+        [["CLASSIC_DAYLIGHT"], ["CLASSIC_DAYLIGHT"]],
+    ],
+)
+async def test_setup_classic_led_ctrl(
+    hass: HomeAssistant,
+    eheimdigital_hub_mock: MagicMock,
+    tankconfig: list[list[str]],
+    mock_config_entry: MockConfigEntry,
+    entity_registry: er.EntityRegistry,
+    snapshot: SnapshotAssertion,
+    classic_led_ctrl_mock: MagicMock,
+) -> None:
+    """Test light platform setup with different channels."""
+    mock_config_entry.add_to_hass(hass)
+
+    classic_led_ctrl_mock.tankconfig = tankconfig
+
+    eheimdigital_hub_mock.return_value.devices = {
+        "00:00:00:00:00:01": classic_led_ctrl_mock
+    }
+    eheimdigital_hub_mock.return_value.main = classic_led_ctrl_mock
+
+    with patch("homeassistant.components.eheimdigital.PLATFORMS", [Platform.LIGHT]):
+        await hass.config_entries.async_setup(mock_config_entry.entry_id)
+    await mock_config_entry.runtime_data._async_device_found(
```

Comment:
We should not touch internals like runtime_data.

Reply:
This just triggers the callback from the library to the coordinator manually, as we don't have a websocket connection which would trigger it normally.

Reply:
You can fetch the callback from the mock and then call it manually. Check `knocki` for an example